software/py/vanilla_pc_histogram.py

```diff
@@ -57,20 +47,27 @@ class PCHistogram:
 
 
     # default constructor
-    def __init__(self, per_tile_stat, input_file):
+    def __init__(self, per_tile_stat, trace_file):
 
         self.per_tile_stat = per_tile_stat
 
-        self.traces = []
-        self.manycore_pc_cnt = Counter()
-        self.tile_pc_cnt = Counter()
-        self.min_pc_val = self._BSG_PC_MIN
-        self.max_pc_val = self._BSG_PC_MAX
+        # Parse operation trace file and extract traces 
+        self.traces, self.manycore_dim_y, self.manycore_dim_x = self.__parse_traces (trace_file)
 
-        unorigin = (0,0)
+        # Generate per tile PC count dictionary
+        self.tile_pc_cnt = self.__generate_tile_pc_cnt(self.traces)
 
-        # parse vanilla_operation_trace.log
-        with open(input_file) as f:
+        # Generate PC count dictionary for the entire network
+        self.manycore_pc_cnt = self.__generate_manycore_pc_cnt(self.tile_pc_cnt)
+
+        return
+
+
+    # parse trace file and extract traces
+    def __parse_traces(self, trace_file):
```

Comment:
If we get to the point where this is running out of memory, it would be good to have this method return an iterator or parse the statistics here, so that the whole file doesn't need to be read into memory.

Reply:
Not entirely sure what you mean. We parse traces this way in all of our profiler tools, blood graph, stats parser and the PC histogram. I think this is subject for another PR right?

Reply:
Yes, this isn't something to fix. Just a note